extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchToStringPostAggregator.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.datasketches.hll;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Ordering;
+import com.yahoo.sketches.hll.HllSketch;
+
+import org.apache.druid.query.aggregation.AggregatorFactory;
+import org.apache.druid.query.aggregation.AggregatorUtil;
+import org.apache.druid.query.aggregation.PostAggregator;
+import org.apache.druid.query.cache.CacheKeyBuilder;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Returns a human-readable summary of a given {@link HllSketch}.
+ * This is a string returned by toString() method of the sketch.
+ * This can be useful for debugging.
+ * @author Alexander Saydakov
+ */
+public class HllSketchToStringPostAggregator implements PostAggregator
+{
+
+  private final String name;
+  private final PostAggregator field;
+
+  @JsonCreator
+  public HllSketchToStringPostAggregator(
+      @JsonProperty("name") final String name,
+      @JsonProperty("field") final PostAggregator field
+  )
+  {
+    this.name = name;
+    this.field = field;
+  }
+
+  @Override
+  public Set<String> getDependentFields()
+  {
+    return field.getDependentFields();
+  }
+
+  @Override
+  public Comparator<String> getComparator()
+  {
+    return Ordering.natural();
```

Comment:
Better to use `Comparator.naturalOrder()`. BTW above in similar situation null-aware comparator is used. I'm not sure if null-awareness is needed.

Reply:
Ordering.natural is used 69 times in the Druid codebase, vs 5 for Comparator.naturalOrder. I think @AlexanderSaydakov could be forgiven for thinking Ordering.natural is the way to go.

Please, keep review comments to substantive design, coding, or style issues, rather than personal preference. If you want to start a discussion creating new style rules for Druid as a whole, a mailing list discussion or github issue is a better forum for doing that.

Reply:
Which way is preferred? I will leave as it is for now.

Reply:
@AlexanderSaydakov `Comparator.naturalOrder()` is preferred because it's suggested as a preferred replacement even in the javadoc of `Ordering.natural()` itself. But there is a minor difference, it's ok not to change.

BTW the nullability question is not clarified

Reply:
@gianm I suspect that most of that 69 usages of `Ordering.natural()` were introduced when Druid had to support Java 7, or after that, simply because `Comparator.naturalOrder()` is lesser known. `Comparator.naturalOrder()` is slightly preferred because it's standard, but the difference is so minor that it's OK to ignore such comments for PR authors. I leave them because PR authors might be unaware of this API, but when they become aware they prefer to use `Comparator.naturalOrder()` themselves.

Reply:
@leventov that is reasonable -- in that case please make it clear in your review comment that it is a mere suggestion. Otherwise a comment like "Better to use Comparator.naturalOrder()." could be interpreted as a required change.

BTW, we have Comparators.naturalNullsFirst() in Druid, which is my preference when you want it to work on nulls and you want those to go first. It is a singleton.